lib/wasmex/components/component.ex

```diff
@@ -31,4 +31,40 @@ defmodule Wasmex.Components.Component do
       resource -> {:ok, __wrap_resource__(resource)}
     end
   end
+
+  defmacro __using__(opts) do
+    genserver_stuff =
+      quote do
+        use GenServer
+
+        def start_link(opts) do
+          Wasmex.Components.start_link(opts)
+        end
+
+        def handle_call(request, from, state) do
+          Wasmex.Components.handle_call(request, from, state)
+        end
+      end
+
+    functions =
+      if wit_path = Keyword.get(opts, :wit) do
+        wit_contents = File.read!(wit_path)
```

Comment:
thinking aloud: should we maybe not hand over a file path but rather the wit definition (file content) itself?
maybe the wit content doesn't come from a file but from - i don't know - a network request or was generated somehow.

Reply:
👍

Reply:
Hmm.. as it turns out rust code annoyingly needs a path passed to it, but it doesn't have to actually be a path that exists. I think it uses it to produce errors perhaps. In any event, if we change it to pass the wit contents as a string we have to make up a path somehow. Any thoughts?

Reply:
I'm thinking I will just have the `wit` option just mean a string containing the wit content, and add a `wit_path` or `wit_file` option to specify the file(s). In the future I'd also like to add `component` or `component_bytes` options to pass in the binary, which I think I can also parse with a little more code